spec/rubocop/config_loader_spec.rb

```diff
@@ -230,6 +230,94 @@
       end
     end
 
+    context 'when inherit_mode is set to merge for Exclude' do
+      let(:file_path) { '.rubocop.yml' }
+
+      before do
+        create_file(file_path, <<-YAML.strip_indent)
+          inherit_from: .rubocop_parent.yml
+          inherit_mode:
+            merge:
+              - Exclude
+
+          Style/For:
+            Exclude:
+              - spec/requests/group_invite_spec.rb
+        YAML
+
+        create_file('.rubocop_parent.yml', <<-YAML.strip_indent)
+          Style/For:
+            Exclude:
+              - 'spec/models/expense_spec.rb'
+              - 'spec/models/group_spec.rb'
+        YAML
+      end
+
+      it 'unions the two lists of Excludes from the parent and child configs ' \
+         'and does not output a warning' do
+        expect do
+          excludes = configuration_from_file['Style/For']['Exclude']
+          expect(excludes.sort)
+            .to eq([File.expand_path('spec/requests/group_invite_spec.rb'),
+                    File.expand_path('spec/models/expense_spec.rb'),
+                    File.expand_path('spec/models/group_spec.rb')].sort)
+        end.not_to output.to_stderr
+      end
+    end
+
+    context 'when inherit_mode on a cop overrides the global setting' do
+      let(:file_path) { '.rubocop.yml' }
+
+      before do
+        create_file(file_path, <<-YAML.strip_indent)
+          inherit_from: .rubocop_parent.yml
+          inherit_mode:
+            merge:
+              - Exclude
+
+          Style/For:
+            Exclude:
+              - spec/requests/group_invite_spec.rb
+
+          Style/Dir:
+            inherit_mode:
+              override:
+                - Exclude
+            Exclude:
+              - spec/requests/group_invite_spec.rb
+
+        YAML
+
+        create_file('.rubocop_parent.yml', <<-YAML.strip_indent)
+          Style/For:
+            Exclude:
+              - 'spec/models/expense_spec.rb'
+              - 'spec/models/group_spec.rb'
+
+          Style/Dir:
+            Exclude:
+              - 'spec/models/expense_spec.rb'
+              - 'spec/models/group_spec.rb'
+        YAML
+      end
+
+      it 'unions the two lists of Excludes from the parent and child configs ' \
+          'for cops that do not override the inherit_mode' do
+        excludes = configuration_from_file['Style/For']['Exclude']
+        expect(excludes.sort)
```

Comment:
The warnings here are expected and I'll update the specs to expect the warning. (updated here: c00e35d)

It will warn here because the `Style/Dir` config is setting the `inherit_mode` back to the default of override for `Exclude`.